src/App.tsx

```diff
@@ -1,72 +1,130 @@
+      setAppliedQuery('');
+    }
+  };
+
+  const handleInputBlur = () => {
+    setTimeout(() => {
+      setIsDropdownVisible(false);
+    }, 200);
+  };
+
+  const handleSuggestionClick = (person: Person) => {
+    setQuery(person.name);
+    setSelected(person);
+    onSelected?.(person);
+    setIsDropdownVisible(false);
+  };
 
   return (
     <div className="container">
       <main className="section is-flex is-flex-direction-column">
+        {/* Title Section */}
         <h1 className="title" data-cy="title">
-          {`${name} (${born} - ${died})`}
+          {selected
+            ? `${selected.name} (${selected.born} - ${selected.died})`
+            : 'No selected person'}
         </h1>
 
-        <div className="dropdown is-active">
+        {/* Search Section */}
+        <div className={`dropdown ${isDropdownVisible ? 'is-active' : ''}`}>
           <div className="dropdown-trigger">
             <input
               type="text"
               placeholder="Enter a part of the name"
+              value={query}
               className="input"
               data-cy="search-input"
+              onChange={handleInputChange}
+              onFocus={handleInputFocus}
+              onBlur={handleInputBlur}
             />
           </div>
 
-          <div className="dropdown-menu" role="menu" data-cy="suggestions-list">
-            <div className="dropdown-content">
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Bernard Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Antone Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Elisabeth Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter de Decker</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Petronella de Decker</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Elisabeth Hercke</p>
+          {/* Suggestions List */}
+          {isDropdownVisible && (
+            <div
+              className="dropdown-menu"
+              role="menu"
+              data-cy="suggestions-list"
+            >
+              <div className="dropdown-content">
+                {filteredPeople.length > 0 ? (
+                  filteredPeople.map(person => (
+                    <div
+                      key={person.name}
```

Comment:
Using `person.name` as a key can lead to issues if names are not unique. Consider using a unique identifier for each person if available, or ensure that `person.name` is unique.